### PR TITLE
cmd: refine command line help messages and error messages

### PR DESCRIPTION
**Pull request by jiangliu**
Refine command line help messages and error messages.

### CODE REVIEW

File: cmd/containerd-nydus-grpc/main.go [hunk 1]
```diff
@@ -44,40 +44,40 @@ func main() {
...
+				return errors.New("failed to generate nydus default configuration")
 			}
 
 			// Once snapshotter's configuration file is provided, parse it and let command line parameters override it.
 			if snapshotterConfigPath != "" {
 				if c, err := config.LoadSnapshotterConfig(snapshotterConfigPath); err == nil {
 					// Command line parameters override the snapshotter's configurations for backwards compatibility
 					if err := config.ParseParameters(flags.Args, c); err != nil {
-						return errors.Wrap(err, "parse parameters")
+						return errors.Wrap(err, "failed to parse commandline options")
```
Comment (changweige) [inline cmd/containerd-nydus-grpc/main.go:55]: In general, we tend to omit the words like `failed to` or `error` in a Golang wrapped errors chain in case there are too many `fail to` in final error message.

File: cmd/containerd-nydus-grpc/main.go [hunk 2]
```diff
@@ -44,40 +44,40 @@ func main() {
...
 					}
 					snapshotterConfig = *c
 				} else {
-					return errors.Wrapf(err, "Failed to load snapshotter's configuration at %q", snapshotterConfigPath)
+					return errors.Wrapf(err, "failed to load snapshotter configuration from %q", snapshotterConfigPath)
```
Comment (changweige) [inline cmd/containerd-nydus-grpc/main.go:59]: ditto

File: internal/flags/flags.go [hunk 3]
```diff
@@ -33,45 +33,50 @@ type Flags struct {
...
 			Name:        "nydusd-config",
 			Aliases:     []string{"config-path"},
-			Usage:       "path to the nydusd configuration",
+			Usage:       "path to nydusd configuration file",
 			Destination: &args.NydusdConfigPath,
 		},
 		&cli.StringFlag{
 			Name:        "daemon-mode",
-			Usage:       "spawning nydusd daemon mode, legal values include \"multiple\", \"shared\" or \"none\"",
+			Usage:       "nydusd daemon working mode, possible values: \"dedicated\"/\"multiple\", \"shared\" or \"none\"",
```
Comment (changweige) [inline internal/flags/flags.go:69]: `dedicated` should not be included in this PR?

**Review by changweige — COMMENTED** [review]
Otherwise, LGTM

File: cmd/containerd-nydus-grpc/main.go (hunk 1 above)
Comment (jiangliu) [reply cmd/containerd-nydus-grpc/main.go:55]: It's at the top most level, so added `failed to`:)

File: internal/flags/flags.go (hunk 3 above)
Comment (jiangliu) [reply internal/flags/flags.go:69]: good catch

File: cmd/containerd-nydus-grpc/main.go (hunk 1 above)
Comment (changweige) [reply cmd/containerd-nydus-grpc/main.go:55]: the returned error is not the topmost level, I am afraid. Its parent level is `err` returned from `ParseParameters()`

File: cmd/containerd-nydus-grpc/main.go [hunk 4]
```diff
@@ -44,40 +44,40 @@ func main() {
...
 			}
 
 			if err := config.ValidateConfig(&snapshotterConfig); err != nil {
-				return errors.Wrapf(err, "validate configurations")
+				return errors.Wrapf(err, "failed to validate configurations")
 			}
 
 			if err := config.ProcessConfigurations(&snapshotterConfig); err != nil {
-				return errors.Wrap(err, "process configurations")
+				return errors.Wrap(err, "failed to process configurations")
```
Comment (changweige) [inline cmd/containerd-nydus-grpc/main.go:76]: ditto

File: cmd/containerd-nydus-grpc/main.go (hunk 1 above)
Comment (changweige) [reply cmd/containerd-nydus-grpc/main.go:55]: True.

**Review by changweige — APPROVED** [review]
LGTM
